src/cascadia/TerminalApp/AppCommandlineArgs.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "ActionAndArgs.h"
+
+#pragma once
+#include "Commandline.h"
+
+// fwdecl unittest classes
+namespace TerminalAppLocalTests
+{
+    class CommandlineTest;
+};
+
+namespace TerminalApp
+{
+    class AppCommandlineArgs;
+};
+
+class TerminalApp::AppCommandlineArgs final
+{
+public:
+    static constexpr std::string_view NixHelpFlag{ "-?" };
+    static constexpr std::string_view WindowsHelpFlag{ "/?" };
+
+    AppCommandlineArgs();
+    ~AppCommandlineArgs() = default;
+    int ParseCommand(const Commandline& command);
+
+    static std::vector<Commandline> BuildCommands(const int argc, const wchar_t* argv[]);
+    static std::vector<Commandline> BuildCommands(winrt::array_view<const winrt::hstring>& args);
+
+    void ValidateStartupCommands();
+    std::deque<winrt::TerminalApp::ActionAndArgs>& GetStartupActions();
+    const std::string& GetExitMessage();
+
+private:
+    static const std::wregex _commandDelimiterRegex;
+
+    CLI::App _app{ "wt - the Windows Terminal" };
+    // --- Subcommands ---
+    CLI::App* _newTabCommand;
+    CLI::App* _newPaneCommand;
+    CLI::App* _focusTabCommand;
+    // Are you adding a new sub-command? Make sure to update _noCommandsProvided!
+
+    std::string _profileName;
+    std::string _startingDirectory;
+    std::vector<std::string> _commandline;
+
+    bool _splitVertical{ false };
+    bool _splitHorizontal{ false };
+
+    int _focusTabIndex{ -1 };
+    bool _focusNextTab{ false };
+    bool _focusPrevTab{ false };
+    // Are you adding more args here? Make sure to reset them in _resetStateToDefault

```

Comment:
should these be wrapped into a nested class with a default constructor so that resetstate can just be `state = {}`?